src/components/amount-input.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+
+interface AmountInputProps {
+  label: string;
+  amount: number | null;
+  currency: string;
+  placeholder: string;
+  onChange: (amount: number | null) => void;
+  maxAmount?: string | null;
+}
+
+const AmountInput: React.FC<AmountInputProps> = ({
+  label,
+  amount,
+  currency,
+  onChange,
+  placeholder,
+  maxAmount,
```

Comment:
probably this prop could be renamed to `errorMessage` or something because by this name it's not clear that input is in error state. but it's not a major issue of course :)